lms/services/vitalsource/_schemas.py

```diff
@@ -30,4 +30,17 @@ class Meta:
         cfi = fields.Str(required=True)
         page = fields.Str(required=True)
 
+        url = fields.Str(required=False)
+        """vitalsource:// like url identifying the book and chapter"""
+
+        @post_load
+        def set_url(self, in_data, **_kwargs):
+            # pylint:disable=import-outside-toplevel
```

Comment:
Avoiding circular import here.

I do think `generate_document_url` belongs to the service so this annotations is probably better that moving the function to some other namespace.